securedrop/tests/test_unit_source.py

```diff
@@ -86,6 +86,17 @@ def test_generate_has_login_link(self):
         already_have_codename_link = soup.select('a#already-have-codename')[0]
         self.assertEqual(already_have_codename_link['href'], '/login')
 
+    def test_generate_already_logged_in(self):
+        self._new_codename()
+        # Make sure it redirects to /lookup when logged in
+        rv = self.client.get('/generate')
+        self.assertEqual(rv.status_code, 302)
+        # Make sure it flashes the message on the lookup page
+        rv = self.client.get('/generate', follow_redirects=True)
+        # Should redirect to /lookup
+        self.assertEqual(rv.status_code, 200)
+        self.assertIn("because you are already logged in.", rv.data)
+
```

Comment:
Haven't gotten a functional test yet, but that should handle the unit test portion of logging out. I'll try to work out the functional tests, but any tips would greatly appreciated.